django/utils/http.py

```diff
@@ -302,7 +302,10 @@ def _urlparse(url, scheme='', allow_fragments=True):
     Note that we don't break the components up in smaller bits
     (e.g. netloc is a single string) and we don't expand % escapes."""
     url, scheme, _coerce_result = _coerce_args(url, scheme)
-    splitresult = _urlsplit(url, scheme, allow_fragments)
+    try:
+        splitresult = _urlsplit(url, scheme, allow_fragments)
+    except ValueError:
+        return False
```

Comment:
This changes the return value of `_urlparse`. I think it would be more appropriate to catch the `ValueError` in `_is_safe_url` instead and return `False` there.